### PR TITLE
Control extensions

### PR DESCRIPTION
**Pull request by abelermias**

### CODE REVIEW

File: controlExtensions/src/main/java/ControlExtensions/RadioGroup.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+public class RadioGroup extends ControlExtension{
+
+	public RadioGroup(WebElement mappedElement) {
+		super(mappedElement);
+	}
+
+	public RadioButton getButton(String label) {
+		
+		WebElement button = this.mappedElement.findElement(By.xpath("//label[text()='" + label + "']/../label"));
```
Comment (dentaylor) [inline controlExtensions/src/main/java/ControlExtensions/RadioGroup.java:14]: xpaths in Control Extensions should be relative. Replace ./ with //

File: controlExtensions/src/main/java/ControlExtensions/RadioGroup.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		return new RadioButton(button);
+	}
+
+	public String getSelected() {
+		
+		WebElement selected = this.mappedElement.findElement(By.xpath("//p/span"));
```
Comment (dentaylor) [inline controlExtensions/src/main/java/ControlExtensions/RadioGroup.java:20]: same

File: controlExtensions/src/main/java/ControlExtensions/Slider.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		sliderValue = mappedElement.findElement(By.id("sliderValue"));
+		
+		int currVal = Integer.parseInt(sliderValue.getAttribute("value"));
+		
+		if(currVal < Integer.parseInt(value)) {
+			while(!sliderValue.getAttribute("value").equals(value))
+				this.slider.sendKeys(Keys.ARROW_RIGHT);
+		}
+		
+		if(currVal > Integer.parseInt(value)) {
```
Comment (dentaylor) [inline controlExtensions/src/main/java/ControlExtensions/Slider.java:28]: use your getValue method here

File: controlExtensions/src/main/java/ControlExtensions/Slider.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	}
+	
+	public void setValue(String value) {
+		
+		sliderValue = mappedElement.findElement(By.id("sliderValue"));
+		
+		int currVal = Integer.parseInt(sliderValue.getAttribute("value"));
+		
+		if(currVal < Integer.parseInt(value)) {
+			while(!sliderValue.getAttribute("value").equals(value))
```
Comment (dentaylor) [inline controlExtensions/src/main/java/ControlExtensions/Slider.java:24]: use your getValue method here

File: controlExtensions/src/test/java/SelectMenuTests.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		Select selectMenu = selectMenuPage.getSelectMenu();
+		selectMenu.selectByVisibleText(expectedText);
+		
+		var actualText = selectMenu.getFirstSelectedOption().getText();
+		
+		assertEquals(actualText, expectedText, "Should return selected text");
+	}
+	
+	@Test
+	public void canGetAllOptions() {
```
Comment (dentaylor) [inline controlExtensions/src/test/java/SelectMenuTests.java:34]: incomplete

File: controlExtensions/src/test/java/RadioButtonTests.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+		this.driver.navigate().to(URL);
+	}
+	
+	@Test
+	public void canSelectYesButton() {
+		
+		var expectedLabel = "Yes";
+		
+		var radioGroup = new RadioButtonPage(this.driver)
+								.navigate()
```
Comment (dentaylor) [inline controlExtensions/src/test/java/RadioButtonTests.java:23]: Fix indentation

**Review by dentaylor — APPROVED** [review]

**Comment by abelermias** [comment]
I did the cucumber project in the wrong branch, apologies

**Review by dentaylor — APPROVED** [review]

**Pull request merged**